common/src/main/java/io/homeassistant/companion/android/common/assist/AssistViewModelBase.kt

```diff
@@ -68,12 +70,12 @@ abstract class AssistViewModelBase(
      * @param pipeline information about the pipeline, or `null` to use the server's default
      * @param onMessage callback for messages that should be posted for this pipeline run, with 3
      * arguments: the message, whether the message is input/output/undetermined, whether the message
-     * is an error message
+     * is an error message, whether the conversation should continue
      */
     protected fun runAssistPipelineInternal(
         text: String?,
         pipeline: AssistPipelineResponse?,
-        onMessage: (String, Boolean?, Boolean) -> Unit
+        onMessage: (String, Boolean?, Boolean, Boolean) -> Unit
```

Comment:
I'm not a fan of this callback:
- When using it it's not clear what are the parameters
- The parameters are not always relevant depending on the event we received
- Nullable boolean is kinda weird (3 states instead of 2 it makes the rest of the logic complicated to understand)

We could explore
- Use Flow instead of a callback
- Encapsulate the events into a `sealed class` to have specifics events with proper attributes depending on the context